### ISSUE TITLE
GraphicsView DrawString not rendering in iOS

### ISSUE THREAD

**Issue by hbraasch** [dotnet/maui#8486]
### Description

The simple repro project (see link below) displays superimposed text from a label and text from a GraphicsView.
It renders correct under Android and Windows, but fails under iOS

Here is the correct render for Android

![AndroidPass](https://user-images.githubusercontent.com/7484186/176988060-eb46b53e-3214-4402-814e-968fe59f97b2.png)

Here is the correct render for Windows

![WinPass](https://user-images.githubusercontent.com/7484186/176988069-36e96398-7e85-490d-9739-b5d4ab68f816.png)

Here is the failed render for iOS

![iOSfail](https://user-images.githubusercontent.com/7484186/176988076-3a4b26fd-e04f-4925-9b70-6a20594fa82c.png)

Here is the link to the repro project

https://github.com/hbraasch/GraphicsViewTester2.git

### Steps to Reproduce

1. Open project from repro link inside VS. Sample code is mainly in file "StartupPage.cs"
2. Run project under Android, see it renders as expected
3. Run project under Windows, see it renders as expected
4. Run project under iOS, see it renders incorrectly

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15.4

### Did you find any workaround?

None

**Comment by jsuarezruiz** [comment]
Hey @hbraasch, it seems that the repository does not exist or is not public?, could you publish it again? https://github.com/hbraasch/GraphicsViewTester2

**Comment by hbraasch** [comment]
Sorry, it was private. I've now changed it to public.
Regards

**Comment by jsuarezruiz** [comment]
Verified with https://github.com/hbraasch/GraphicsViewTester2

**Comment by jsuarezruiz** [comment]
```
  at CoreGraphics.CGFont.Create (System.IntPtr handle) [0x00010] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/CoreGraphics/CGFont.cs:59 
  at CoreGraphics.CGFont.CreateWithFontName (System.String name) [0x00015] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/CoreGraphics/CGFont.cs:102 
  at Microsoft.Maui.Graphics.Platform.FontExtensions.GetDefaultCGFont () [0x00000] in /Users/jsuarez/Documents/GitHub/System.Graphics/src/Microsoft.Maui.Graphics/MaciOS/FontExtensions.cs:54 
  at Microsoft.Maui.Graphics.Platform.FontExtensions.ToCGFont (Microsoft.Maui.Graphics.IFont font) [0x00000] in /Users/jsuarez/Documents/GitHub/System.Graphics/src/Microsoft.Maui.Graphics/MaciOS/FontExtensions.cs:16 
  at Microsoft.Maui.Graphics.Platform.PlatformCanvas.DrawString (System.String value, System.Single x, System.Single y) [0x00013] in /Users/jsuarez/Documents/GitHub/System.Graphics/src/Microsoft.Maui.Graphics/MaciOS/PlatformCanvas.cs:1010 
  at Microsoft.Maui.Graphics.Platform.PlatformCanvas.DrawString (System.String value, System.Single x, System.Single y, Microsoft.Maui.Graphics.HorizontalAlignment horizontalAlignment) [0x0000b] in /Users/jsuarez/Documents/GitHub/System.Graphics/src/Microsoft.Maui.Graphics/MaciOS/PlatformCanvas.cs:991 
  at GraphicsTester.Scenarios.DrawTextAtPoint.Draw (Microsoft.Maui.Graphics.ICanvas canvas) [0x0005d] in /Users/jsuarez/Documents/GitHub/System.Graphics/samples/GraphicsTester.Portable/Scenarios/DrawTextAtPoint.cs:21 
  at GraphicsTester.Scenarios.AbstractScenario.Draw (Microsoft.Maui.Graphics.ICanvas canvas, Microsoft.Maui.Graphics.RectF dirtyRect) [0x00001] in /Users/jsuarez/Documents/GitHub/System.Graphics/samples/GraphicsTester.Portable/Scenarios/AbstractScenario.cs:69 
  at Microsoft.Maui.Graphics.Platform.DirectRenderer.Draw (CoreGraphics.CGContext coreGraphics, Microsoft.Maui.Graphics.RectF dirtyRect) [0x0000f] in /Users/jsuarez/Documents/GitHub/System.Graphics/src/Microsoft.Maui.Graphics/Mac/DirectRenderer.cs:39 
```

**Comment by pengalo** [comment]
Ran into this same issue today and haven't found a work around. Microsoft.Maui.Graphics version 6.0.403, iOS 15.6.1. Renders text on Android, not on iOS.

**Comment by hbraasch** [comment]
Not fixing it, but maybe a temporary workaround I've found...


Specify the rectangle when drawing the text. Like this:
```
 canvas.DrawString("Text", 0 0, dirtyRect.Width, dirtyRect.Height, HorizontalAlignment.Left, VerticalAlignment.Top, textFlow: TextFlow.OverflowBounds);
```

The [DrawString] without the rectangle does not render under iOS:
```
canvas.DrawString("Text", 0, 0, HorizontalAlignment.Left);
```

**Comment by jinxinjuan** [comment]
Verified this issue with Visual Studio Enterprise 17.6.0 Preview 7.0. Repro on iOS platform with project.[GraphicsViewTester2.git](https://github.com/hbraasch/GraphicsViewTester2.git)

**Comment by jsuarezruiz** [comment]
Cannot reproduce with the latest version
<img width="341" alt="Captura de pantalla 2023-07-06 a las 9 01 21" src="https://github.com/dotnet/maui/assets/6755973/5656d2fb-86a9-4094-a3cf-444dcaf920ac">

**Comment by hartez** [comment]
@jsuarezruiz I'm running into this on 8 preview 6 - did something happen since then to fix it?

**Comment by malsabi** [comment]
It's not working on RC2 .NET 8 Core

**Comment by malsabi** [comment]
@jsuarezruiz I have used Progress Radial code from Aloha Kit in RC2 .NET 8 Core, the DrawString is completely not showing, I have tried to provide width, height but still not working at all. Is there going to be a fix in the GA release atleast ?

**Comment by hartez** [comment]
The behavior I was seeing was that DrawString will work if the resulting text fits in the target rectangle. If it's too large, then instead of clipping the text, the text just disappears entirely.

**Comment by malsabi** [comment]
> The behavior I was seeing was that DrawString will work if the resulting text fits in the target rectangle. If it's too large, then instead of clipping the text, the text just disappears entirely.

Yes you're right 100% even if it exceeds the rectangle a single pixel it will disappear entirely it should be within the rectangle.

**Comment by riccardominato** [comment]
This problem verifies even trying to calculate the exact bounds using `canvas.GetStringSize()`.

```csharp
public void Draw(ICanvas canvas, RectF dirtyRect)
{
    var stringSize = canvas.GetStringSize("HELLO, WORLD!", Font.Default, 20);

    var stringBounds = new RectF(new PointF(0, 0), stringSize);
    
    canvas.StrokeColor = Colors.Red;
    canvas.StrokeSize = 2;
    canvas.DrawRectangle(stringBounds);
    
    canvas.FontSize = 20;

    canvas.DrawString("HELLO, WORLD!", stringBounds, HorizontalAlignment.Left, VerticalAlignment.Top);
}
```

In this specific case, I can solve by slightly increasing bounds height.

```csharp
var stringBounds = new RectF(new PointF(0, 0), new Size(stringSize.Width, stringSize.Height + 2));
```

**Comment by riccardominato** [comment]
There are many open issues related to this one, which is quite old.
Personally, I still have troubles caused by https://github.com/dotnet/maui/issues/18679.

Could you please take a look to understand why this is happening and fix it?

Thank you.

**Comment by ghost** [comment]
Until the fix is completed, please find [workaround here](https://github.com/dotnet/maui/issues/4993#issuecomment-3101239999).
